src/service/history-impl.js

```diff
@@ -899,17 +902,27 @@ export class HistoryBindingVirtual_ {
    * otherwise.
    * @param {*} maybeHistoryState
    * @param {!HistoryStateDef} fallbackState
+   * @param {string} debugId
    * @return {!HistoryStateDef}
    * @private
    */
-  toHistoryState_(maybeHistoryState, fallbackState) {
-    if (maybeHistoryState && maybeHistoryState['stackIndex']) {
+  toHistoryState_(maybeHistoryState, fallbackState, debugId) {
+    if (this.isHistoryState_(maybeHistoryState)) {
       return /** @type {!HistoryStateDef} */ (maybeHistoryState);
+    } else {
+      dev().warn(TAG_, BAD_DATA_WARNING, debugId, maybeHistoryState);
```

Comment:
I think the message has to be inline for the message extractor to work.

Reply:
Done.